packages/core/src/messageStore.ts

```diff
@@ -199,65 +188,16 @@ export class MessageStore {
 		}
 	}
 
-	private getBlockId(block: Block): number {
-		const record: undefined | { id: number; blockhash: Buffer; timestamp: number } = this.statements.getBlockId.get({
-			chain: block.chain,
-			chain_id: block.chainId,
-			blocknum: block.blocknum,
-		})
-
-		if (record === undefined) {
-			const { lastInsertRowid } = this.statements.insertBlock.run({
-				chain: block.chain,
-				chain_id: block.chainId,
-				blockhash: fromHex(block.blockhash),
-				blocknum: block.blocknum,
-				timestamp: block.timestamp,
-			})
-
-			return Number(lastInsertRowid)
-		} else {
-			assert(block.blockhash === toHex(record.blockhash), "MessageStore.getBlockId: blockhashes did not match")
-			assert(block.timestamp === record.timestamp, "MessageStore.getBlockId: timestamps did not match")
-			return record.id
-		}
-	}
-
-	private getBlock(blockId: number): Block {
-		const record: undefined | BlockRecord = this.statements.getBlock.get({ id: blockId })
-		if (record === undefined) {
-			throw new Error("internal error: failed to look up block id")
-		}
-
-		assert(chainType.is(record.chain), `invalid chain type ${JSON.stringify(record.chain)}`)
-
-		return {
-			chain: record.chain,
-			chainId: record.chain_id,
-			blocknum: record.blocknum,
-			blockhash: toHex(record.blockhash),
-			timestamp: record.timestamp,
-		}
-	}
-
-	private static createBlocksTable = `CREATE TABLE IF NOT EXISTS blocks (
-    id        INTEGER PRIMARY KEY AUTOINCREMENT,
-    chain     TEXT    NOT NULL,
-    chain_id  INTEGER NOT NULL,
-    blocknum  INTEGER NOT NULL,
-    blockhash BLOB    NOT NULL,
-    timestamp INTEGER NOT NULL,
-    UNIQUE(chain, chain_id, blocknum)
-  );`
-
 	private static createActionsTable = `CREATE TABLE IF NOT EXISTS actions (
     id              INTEGER PRIMARY KEY AUTOINCREMENT,
     hash            BLOB    NOT NULL UNIQUE,
     signature       BLOB    NOT NULL,
     session_address BLOB    REFERENCES sessions(session_address),
     from_address    BLOB    NOT NULL,
     timestamp       INTEGER NOT NULL,
-    block_id        INTEGER REFERENCES blocks(id),
+    blockhash       BLOB    NOT NULL,
```

Comment:
oh you already have it typed as blob... I think this is probably ultimately ignored because sqlite is untyped by default lol

Reply:
so changing the type in the Record type to `Buffer` was the right move